src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs

```diff
@@ -49,6 +49,23 @@ public JoinFederationRequestService(ICounterChainSettings counterChainSettings,
             this.votingManager = votingManager;
         }
 
+        public static byte[] GetFederationMemberBytes(JoinFederationRequest joinRequest, Network network, Network counterChainNetwork)
+        {
+            Script collateralScript = PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(joinRequest.CollateralMainchainAddress);
+            BitcoinAddress bitcoinAddress = collateralScript.GetDestinationAddress(counterChainNetwork);
+            var collateralFederationMember = new CollateralFederationMember(joinRequest.PubKey, false, joinRequest.CollateralAmount, bitcoinAddress.ToString());
+
+            return (network.Consensus.ConsensusFactory as CollateralPoAConsensusFactory).SerializeFederationMember(collateralFederationMember);
+        }
+
+        public static void SetLastRemovalEventId(JoinFederationRequest joinRequest, byte[] federationMemberBytes, VotingManager votingManager)
+        {
+            Poll poll = votingManager.GetApprovedPolls().OrderByDescending(p => p.PollStartBlockData.Height).FirstOrDefault(x => x.IsExecuted &&
```

Comment:
You can use `this.votingManager.GetExecutedPolls()` here rather :)

Reply:
Changed.